q3tiledaligner/src/au/edu/qimr/tiledaligner/util/TARecordUtil.java

```diff
@@ -483,6 +132,97 @@ public static TLongList getLongList(long ... list) {
+			
+			List<BLATRecord[]> blats = new ArrayList<>(maxSplits.size() + 1);
+			for (IntLongPairs maxSplitILPs : maxSplits) {
+				BLATRecord [] blatties = null;
+				/*
+				 * will attempt to create a single BLAT record
+				 */
+				if (IntLongPairsUtil.isIntLongPairsAValidSingleRecord(maxSplitILPs)) {
+					String [] blatData = TARecordUtil.blatRecordFromSplits(maxSplitILPs, name, seqLength, headerMap, TILE_LENGTH);
+					if (null != blatData && blatData.length > 0) {
+						blatties = new BLATRecord[] {new BLATRecord(blatData)};
+					}
+				} else {
+					
+					IntLongPair[] pairs = maxSplitILPs.getPairs();
+					logger.info("createing BLAT records, number of constituent ILPs in ILPS: " + pairs.length);
+					if (pairs.length < 3) {
+						/*
+						 * return a BLAT record for each constituent in the maxSplitILPs
+						 */
+						blatties = new BLATRecord[pairs.length];
+						for (int i = 0 ; i < pairs.length ; i++) {
+							String [] blatARray = blatRecordFromSplit(pairs[i], name, seqLength, headerMap);
+							if (null != blatARray && blatARray.length > 0) {
+								blatties[i] = new BLATRecord(blatARray);
+							}
+						}
+						Arrays.sort(blatties);
+					} else {
+						
+						/*
+						 * from constituent ILPs, find largst, and then see if any combination of adding other ILPs results in a single BLAT record
+						 */
+						Optional<IntLongPairs> oSingleBLATRec = IntLongPairsUtil.getSingleBLATRecordFromILPs(maxSplitILPs);
+						if (oSingleBLATRec.isPresent()) {
+							/*
+							 * need to find the ILPs that didn't make it into the ILPS so that they can be added as separate BLAT records
+							 */
+							List<IntLongPair> rejectedILPs = IntLongPairsUtil.getRejectedILPs(maxSplitILPs, oSingleBLATRec.get());
+							logger.info("found optional singleBLATRecord! Number of rejected ILPs: " + rejectedILPs.size());
+							
+							blatties = new BLATRecord[rejectedILPs.size() + 1];
+							for (int i = 0 ; i < rejectedILPs.size() ; i++) {
+								String [] blatArray = blatRecordFromSplit(rejectedILPs.get(i), name, seqLength, headerMap);
+								if (null != blatArray && blatArray.length > 0) {
+									blatties[i] = new BLATRecord(blatArray);
+								}
+							}
+							String [] blatArray = blatRecordFromSplits(oSingleBLATRec.get(), name, seqLength, headerMap, TILE_LENGTH);	
+							if (null != blatArray && blatArray.length > 0) {
+								blatties[blatties.length - 1] = new BLATRecord(blatArray);
+							}
+							
+							if (null != blatties && blatties.length > 1) {
+								Arrays.sort(blatties);
+							}
+						}
+					}
+				}
+				if (null != blatties && blatties.length > 0) {
+					blats.add(blatties);
+				}
+			}
+			return blats;
+		}
+		return Collections.emptyList();
+	}
+	
 	public static String[] blatRecordFromSplit(IntLongPair split, String name, int seqLength, PositionChrPositionMap headerMap) {
```

Comment:
A bit pedantic perhaps, but the name of this implies it returns BLATRecord - is there a reason we're returning the raw String array and not the constructed BLATRecord?